### PR TITLE
Merge official and external third party exporter-listing

### PR DESCRIPTION
**Pull request by cherti**
This merges the official third party exporters and the external ones into one category structure. the official ones are, as proposed by @juliusv, marked.

The surrounding text was adapted as well to reflect the new structure.

### CODE REVIEW

**Comment by juliusv** [comment]
Ah, could you do bold instead of italic for the official ones? Stands out more...

File: content/docs/instrumenting/exporters.md [hunk 1]
```diff
@@ -71,12 +57,14 @@ hosted outside of the Prometheus GitHub organization.
...
 ### HTTP
    * [Apache exporter](https://github.com/neezgee/apache_exporter)
+   * [HAProxy exporter](https://github.com/prometheus/haproxy_exporter) (*official*)
    * [Nginx metric library](https://github.com/knyar/nginx-lua-prometheus)
    * [Passenger exporter](https://github.com/stuartnelson3/passenger_exporter)
    * [Varnish exporter](https://github.com/jonnenauha/prometheus_varnish_exporter)
    * [WebDriver exporter](https://github.com/mattbostock/webdriver_exporter)
    
 ### APIs
+   * [AWS CloudWatch exporter](https://github.com/prometheus/cloudwatch_exporter) (*official*)
```
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: The CloudWatch exporter could also count as another monitoring system.

File: content/docs/instrumenting/exporters.md [hunk 2]
```diff
@@ -89,22 +77,32 @@ hosted outside of the Prometheus GitHub organization.
...
+   * [StatsD exporter](https://github.com/prometheus/statsd_exporter) (*official*)
+
 
 ### Miscellaneous
    * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
    * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [Blackbox exporter](https://github.com/prometheus/blackbox_exporter) (*official*)
    * [Jenkins exporter](https://github.com/lovoo/jenkins_exporter)
+   * [JMX exporter](https://github.com/prometheus/jmx_exporter) (*official*)
+   * [Memcached exporter](https://github.com/prometheus/memcached_exporter) (*official*)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: If redis is a database, so is this

File: content/docs/instrumenting/exporters.md [hunk 3]
```diff
@@ -89,22 +77,32 @@ hosted outside of the Prometheus GitHub organization.
...
+   * [Mesos task exporter](https://github.com/prometheus/mesos_exporter) (*official*)
    * [Meteor JS web framework exporter](https://atmospherejs.com/sevki/prometheus-exporter)
    * [Minecraft exporter module](https://github.com/Baughn/PrometheusIntegration)
    * [PowerDNS exporter](https://github.com/janeczku/powerdns_exporter)
    * [rTorrent exporter](https://github.com/mdlayher/rtorrent_exporter)
    * [SMTP/Maildir MDA blackbox prober](https://github.com/cherti/mailexporter)
+   * [SNMP exporter](https://github.com/prometheus/snmp_exporter) (*official*)
```
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: Not sure about SNMP, but could conceivably also be classified under "Other monitoring systems"? What does @brian-brazil say?

File: content/docs/instrumenting/exporters.md [hunk 4]
```diff
@@ -89,22 +77,32 @@ hosted outside of the Prometheus GitHub organization.
...
    * [Xen exporter](https://github.com/lovoo/xenstats_exporter)
+
```
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: Remove extra empty line.

File: content/docs/instrumenting/exporters.md [hunk 5]
```diff
@@ -89,22 +77,32 @@ hosted outside of the Prometheus GitHub organization.
...
+   * [scollector exporter](https://github.com/tgulacsi/prometheus_scollector)
+   * [StatsD exporter](https://github.com/prometheus/statsd_exporter) (*official*)
+
 
 ### Miscellaneous
    * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
    * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [Blackbox exporter](https://github.com/prometheus/blackbox_exporter) (*official*)
    * [Jenkins exporter](https://github.com/lovoo/jenkins_exporter)
+   * [JMX exporter](https://github.com/prometheus/jmx_exporter) (*official*)
```
Comment (juliusv) [inline content/docs/instrumenting/exporters.md]: Dito for JMX...

File: content/docs/instrumenting/exporters.md (hunk 2 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Indeed. This to the databases and redis there as well?

File: content/docs/instrumenting/exporters.md (hunk 3 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: the same :)

File: content/docs/instrumenting/exporters.md (hunk 5 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: I don't get the context, could you elaborate to which of your comments the "Dito" is referring?

File: content/docs/instrumenting/exporters.md (hunk 2 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: sorted memcached to databases. correct me if both don't fit in there. :)

**Comment by cherti** [comment]
@juliusv @brian-brazil 
everything should be implemented, except for @juliusv comment on the jmx-exporter as I'm not absolutely sure what he's referring to.

File: content/docs/instrumenting/exporters.md [hunk 6]
```diff
@@ -88,17 +76,25 @@ hosted outside of the Prometheus GitHub organization.
...
+   * [SNMP exporter](https://github.com/prometheus/snmp_exporter) (**official**)
+   * [StatsD exporter](https://github.com/prometheus/statsd_exporter) (**official**)
 
 ### Miscellaneous
    * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
    * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [Blackbox exporter](https://github.com/prometheus/blackbox_exporter) (**official**)
    * [Jenkins exporter](https://github.com/lovoo/jenkins_exporter)
+   * [JMX exporter](https://github.com/prometheus/jmx_exporter) (**official**)
+   * [Mesos task exporter](https://github.com/prometheus/mesos_exporter) (**official**)
```
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Then I'll drop it altogether

**Comment by cherti** [comment]
@juliusv @brian-brazil @sdurrheimer 
Anything more to do before this can get merged? It's lying around here for quite a while now without much activity.

File: content/docs/instrumenting/exporters.md [hunk 7]
```diff
@@ -10,42 +10,28 @@ metrics from third-party systems as Prometheus metrics. This is useful for
...
-[Cassandra](http://cassandra.apache.org/).
+Some of these exporters are maintained as part of the official [Prometheus GitHub organization](https://github.com/prometheus),
+those are marked as *official*, others are externally contributed and maintained.
 
-## Independently maintained third-party exporters
-
-There are also a number of exporters which are externally contributed
-and maintained. We encourage the creation of more exporters but cannot
+We encourage the creation of more exporters but cannot
 vet all of them for best practices. Commonly, those exporters are
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: maybe link to the practices here?

File: content/docs/instrumenting/exporters.md [hunk 8]
```diff
@@ -10,42 +10,28 @@ metrics from third-party systems as Prometheus metrics. This is useful for
...
+The [JMX exporter](https://github.com/prometheus/jmx_exporter) can export from a
+wide variety of JVM-based applications, for example [Kafka](http://kafka.apache.org/) and
+[Cassandra](http://cassandra.apache.org/).
+
 ### Databases
    * [Aerospike exporter](https://github.com/alicebob/asprom)
    * [ClickHouse exporter](https://github.com/f1yegor/clickhouse_exporter)
+   * [Consul exporter](https://github.com/prometheus/consul_exporter) (**official**)
    * [CouchDB exporter](https://github.com/gesellix/couchdb-exporter)
+   * [InfluxDB exporter](https://github.com/prometheus/influxdb_exporter) (**official**)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: Influxdb is a monitoring system

File: content/docs/instrumenting/exporters.md [hunk 9]
```diff
@@ -90,18 +78,25 @@ hosted outside of the Prometheus GitHub organization.
...
+   * [SNMP exporter](https://github.com/prometheus/snmp_exporter) (**official**)
+   * [StatsD exporter](https://github.com/prometheus/statsd_exporter) (**official**)
 
 ### Miscellaneous
    * [BIG-IP exporter](https://github.com/ExpressenAB/bigip_exporter)
    * [BIND exporter](https://github.com/digitalocean/bind_exporter)
+   * [Blackbox exporter](https://github.com/prometheus/blackbox_exporter) (**official**)
    * [BOSH exporter](https://github.com/cloudfoundry-community/bosh_exporter)
    * [Jenkins exporter](https://github.com/lovoo/jenkins_exporter)
+   * [JMX exporter](https://github.com/prometheus/jmx_exporter) (**official**)
```
Comment (brian-brazil) [inline content/docs/instrumenting/exporters.md]: I believe Julius was saying that JMX belongs under monitoring systems.

File: content/docs/instrumenting/exporters.md (hunk 7 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: Done. [This one](https://prometheus.io/docs/instrumenting/writing_exporters/) you ment? Just that we avoid miscommunication here.

File: content/docs/instrumenting/exporters.md (hunk 8 above)
Comment (cherti) [inline content/docs/instrumenting/exporters.md]: done

**Comment by cherti** [comment]
@grobie 
Currently the sections are more or less sorted like "if I have seen enough exporters of a kind to justify a category, I'll add it".

Sorting alphabetically would be a possibility, but I think that we should have a special case for *miscellaneous* then, because having the miscellanii in the middle somewhere would be alphabetically correct but I'd find it strange, because *miscellaneous* is basically a catch-all that gets everything we can't sort anywhere other.

**Comment by cherti** [comment]
On the other hand, using alphabetical sorting here might be not too useful, as the category-names are even more "chosen" than the exporter-names. So it might not be too helpful regarding the structure, although certainly possible.

**Comment by grobie** [comment]
I'm fine keeping miscellaneous at the bottom. While it's true that a category name might not be what I had in mind initially, I'll get at least used to them and at that point it becomes easier to scan for an exporter in the list.

**Comment by cherti** [comment]
@grobie 
> While it's true that a category name might not be what I had in mind initially

What did you have in mind initially?

I can try to come up with a reordering-scheme if desired, maybe we have something better than alphabetically (like topic-ish or something). Question is then whether we tackle that in this PR or leave it to a separate one to clean up the diff.

**Comment by grobie** [comment]
I don't have any better suggestions for the current group names. I guess I'd remove "Other" from "Monitoring systems", but it might be clearer that way.

Let's merge and iterate on it.

**Comment by cherti** [comment]
Agreed.

The merge of the official exporters into the categories is done here, I'd say, so I'd consider this one ready for merge.

**Pull request merged**